validations/src/validations.rs

```diff
@@ -509,7 +511,16 @@ pub fn validate_data_request_output(
     }
 
     if active_wips.wip0022() {
-        let reward_collateral_ratio = request.collateral / request.witness_reward;
+        let collateral = if request.collateral == 0 {
+            collateral_minimum
+        } else {
+            request.collateral
+        };
+        let reward_collateral_ratio = if request.witness_reward > 0 {
+            collateral / request.witness_reward
+        } else {
+            u64::MAX
+        };
```

Comment:
Code repetition. I'd try to have a pure function with a `Fn(u64, u64) -> u64` contract that derives this, cover it with some tests, and use it here and anywhere else.

Reply:
Feels like a bit of overkill for an if-statement which occurs only twice, but sure.

Reply:
My personal take is that the more consensus critical logic we can have in "library code" (i.e. not tied to specific data structures or error types), the better for unit testing and for reusing the same implementation across different components.

Reply:
Done. Moved it into a function at `data_request.rs`.